### ISSUE TITLE
LAPACKE issue

### ISSUE THREAD

**Issue by hhadian** [arrayfire/arrayfire#2441]
I have built LAPACK from source (lapack 3.8.0) and when I use it to build arrayfire, it fails with this error:
```
...
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `spotrf_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `dtrtrs_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `zpotrf_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `dgesvd_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `dgetrs_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `sgeqrf_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `sorgqr_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `zgesvd_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `sgesvd_'
../src/backend/cpu/libafcpu.so.3.7.0: undefined reference to `sgetri_'
collect2: error: ld returned 1 exit status
test/CMakeFiles/test_ycbcr_rgb_cpu.dir/build.make:97: recipe for target 'test/ycbcr_rgb_cpu' failed
make[2]: *** [test/ycbcr_rgb_cpu] Error 1
CMakeFiles/Makefile2:2621: recipe for target 'test/CMakeFiles/test_ycbcr_rgb_cpu.dir/all' failed
make[1]: *** [test/CMakeFiles/test_ycbcr_rgb_cpu.dir/all] Error 2
```
I imagine this is related to lapack. Is that right?

I configured cmake using these options for lapack: `-DLAPACK_INCLUDE_DIR=/export/b08/hhadian/lapack-3.8.0/LAPACKE/include/ -DLAPACK_LIBRARIES=/export/b08/hhadian/lapack-3.8.0/liblapacke.a`

Should I provide something else for these options?
Thanks!

**Comment by 9prady9** [comment]
I believe that have something to do with the fact that you are using a static library and we don't have appropriate flags set. I haven't really tried building with static lapacke earlier but I think that's what this is. Can you try shared library? That well at least tell us if this problem is with the library or arrayfire

**Comment by hhadian** [comment]
OK, I compiled `lapacke` shared libraries but there were still undefined references. 
I had to use both `lapack` and `lapacke` to get rid of [most of] those errors. 

There were still two undefined references (to `LAPACKE_dsytrf_aa_work` and `LAPACKE_ssytrf_aa_work`) which were due to a bug in lapack 3.8.0 make files (2 source files were not listed).
Anyway I could successfully build Arrayfire.

Thanks
